packages/wallet/test/rollup-block-submitter.spec.ts

```diff
@@ -0,0 +1,415 @@
+
+      await blockSubmitter.submitBlock(rollupBlock)
+
+      dummyContract.blocksSubmitted.length.should.equal(1)
+      dummyContract.blocksSubmitted[0][0].should.equal(
+        rollupBlock.transitions[0].stateRoot
+      )
+
+      blockSubmitter.getLastConfirmed().should.equal(0)
+      blockSubmitter.getLastSubmitted().should.equal(1)
+      blockSubmitter.getLastQueued().should.equal(1)
+
+      const lastQueuedFromDB: number = await getLastQueuedFromDB(db)
+      lastQueuedFromDB.should.equal(1)
+
+      const lastSubmittedFromDB: number = await getLastSubmittedFromDB(db)
+      lastSubmittedFromDB.should.equal(1)
+    })
+
+    it('should submit new block with one previous block', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        1,
+        1,
+        1,
+        [rollupBlock]
+      )
+
+      await blockSubmitter.submitBlock(rollupBlock2)
+
+      dummyContract.blocksSubmitted.length.should.equal(1)
+      dummyContract.blocksSubmitted[0][0].should.equal(
+        rollupBlock2.transitions[0].stateRoot
+      )
+
+      blockSubmitter.getLastConfirmed().should.equal(1)
+      blockSubmitter.getLastSubmitted().should.equal(2)
+      blockSubmitter.getLastQueued().should.equal(2)
+
+      const lastQueuedFromDB: number = await getLastQueuedFromDB(db)
+      lastQueuedFromDB.should.equal(2)
+
+      const lastSubmittedFromDB: number = await getLastSubmittedFromDB(db)
+      lastSubmittedFromDB.should.equal(2)
+    })
+
+    it('should ignore old block', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        1,
+        1,
+        1,
+        [rollupBlock]
+      )
+
+      await blockSubmitter.submitBlock(rollupBlock)
+
+      dummyContract.blocksSubmitted.length.should.equal(0)
+
+      blockSubmitter.getLastConfirmed().should.equal(1)
+      blockSubmitter.getLastSubmitted().should.equal(1)
+      blockSubmitter.getLastQueued().should.equal(1)
+    })
+
+    it('should queue block when there is one pending', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        1,
+        1,
+        0,
+        [rollupBlock]
+      )
+
+      dummyContract.blocksSubmitted.length = 0
+
+      await blockSubmitter.submitBlock(rollupBlock2)
+
+      dummyContract.blocksSubmitted.length.should.equal(0)
+
+      blockSubmitter.getLastConfirmed().should.equal(0)
+      blockSubmitter.getLastSubmitted().should.equal(1)
+      blockSubmitter.getLastQueued().should.equal(2)
+
+      const lastQueuedFromDB: number = await getLastQueuedFromDB(db)
+      lastQueuedFromDB.should.equal(2)
+    })
+  })
+
+  describe('handleNewRollupBlock()', () => {
+    it('should do nothing when there are no pending blocks', async () => {
+      await blockSubmitter.init()
+
+      await blockSubmitter.handleNewRollupBlock(1)
```

Comment:
Still slightly confused by this function--why is `1` passed here?  Is this simulating a multi-aggregator world where block 1 was submitted by someone else?

Reply:
This is simulating getting a new block event from the contract that says "rollup block 1 was mined"